### PR TITLE
Param to set mime type without guessing.

### PR DESCRIPTION
**Pull request by galachad**
The major change of this pull request is to add the optional `mime_type` to the `attachment` function.

Without this change, you are stuck with `mime::guess_type` function.

### CODE REVIEW

**Comment by datawookie** [comment]
Hi @galachad, I like this idea. A quick question though: do you have examples of where `mime::guess_type()` performs poorly? It'd be helpful to build some test cases around those. Best regards, Andrew.

**Comment by galachad** [comment]
> Hi @galachad, I like this idea. A quick question though: do you have examples of where `mime::guess_type()` performs poorly? It'd be helpful to build some test cases around those. Best regards, Andrew.
Hi @datawookie,

The `mime::guess_type` works great, but it's a poor function base on the file extension.
For example `pdf_file` by default is detected as `text/plain`. Usually, it's not needed but I want to have this possibility to play with mime due to some restricted SMTP service rules (virus scanner etc.).

I also find the bug if you want to use the file without ext.
`guess_type(path, empty = NULL)`
Throwing error
`Error in type[file == ""] <- empty : replacement has length zero`
 
In my opinion, you should change `empty` param from `NULL` to `application/octet-stream`

Best regards, Adam

File: R/attachment.R [hunk 1]
```diff
@@ -12,10 +13,11 @@
 #' attachment(msg, "cat.png")
 #' attachment(msg, "visualisations.png", "visuals")
 #' }
-attachment <- function(msg, path, cid = NA, disposition = NA){
+attachment <- function(msg, path, cid = NA, disposition = NA, mime_type){
   if (length(path) != 1) stop("Must be precisely one attachment.", call. = F)
 
   type <- guess_type(path, empty = NULL)
```
Comment (galachad) [inline R/attachment.R:19]:
```suggestion
  type <- guess_type(path, empty = "application/octet-stream")
```
Corresponding to the https://github.com/datawookie/emayili/pull/22#issuecomment-562022727 If you agree for this change, we can add change in this PR.

**Comment by datawookie** [comment]
Hi @galachad, I've made these changes on the `mime-type` branch. Please can you test and let me know if they satisfy your requirements? Thanks, Andrew.

**Comment by galachad** [comment]
It looks wonderful. You can merge the `mime-type` branch and close this PR. Best regards, Adam

**Comment by datawookie** [comment]
Great. I'm just waiting on feedback for some other features and I'll push a new version of the package later today.

**Pull request closed without merging**